ngx_http_vod_module.c

```diff
@@ -2764,7 +2764,10 @@ ngx_http_vod_run_state_machine(ngx_http_vod_ctx_t *ctx)
 
 		// enable directio if enabled in the configuration (ignore errors)
 		// Note that directio is set on transfer only to allow the kernel to cache the "moov" atom
-		if (ctx->submodule_context.conf->request_handler != ngx_http_vod_remote_request_handler)
+		if (ctx->submodule_context.conf->request_handler == ngx_http_vod_local_request_handler ||
+			(ctx->submodule_context.conf->request_handler == ngx_http_vod_mapped_request_handler &&
+			 ctx->submodule_context.conf->remote_upstream_location.len == 0)
+			)
```

Comment:
This has become a bit complicated, so I think it would be better to add a function called `enable_directio` to ngx_http_vod_reader_t, set it to ngx_file_reader_enable_directio in reader_file... and to null in reader_http. Once you do that, you can just check if ctx->reader->enable_directio is not null